### DISCUSSION TITLE
Vocabulary Builder FAQ

### DISCUSSION THREAD

**Discussion by mergen3107** [koreader/koreader#9256] [Q&A]
This isn't an issue or feature request, I am just wondering about new functionality.

Thank you very much for the Vocabulary Builder! I just discovered I have about 100 pages of look ups throughout many books, from which 85 are actual words :D 

Now, I am a little confused about how to properly use it in this context.

Q1: How should it work in the first place? Do I have to press the Forgot/Got it button for each word? Will flashcards then appear so I can repeat words?

Q2: Is there a way to show words only from specific book/books?

Q3: Is there a way to sort words by hits/misses, alphabet?

**Comment by poire-z** [comment]
Pinging @weijiuqiao (because less chance of being noticed once #9254 is crapverted to a "discussion").

**Comment by weijiuqiao** [comment]
Thanks for the notice. I made this plugin copying ideas from other places, so let me briefly explain how it works. It actually really just has the very basic functions of a flashcard app, a flash list in fact, but it works with the same logic.

This plugin is intended to help you remember words you looked up and want to remember. The ideal use case is like this: You looked up some words during reading, then come to the plugin to check how many you got/forgot by tapping the buttons and delete words that accidentally end up there. After a certain period of time, you come here again to check how much you still remember. With each "got it" for a word, the planned reviewable time is further into the future, and after correctly getting a word for a certain number of times, you are encouraged to delete it.

The long list of words you see is the plugin automatically loading your lookup history the first time it's on your device. It could be overwhelming and kind of defeats the idea of remembering while reading along. So my advice is to just delete all words within the plugin and start anew if you want to try using it. The auto importing is here because some people might want to keep words they have looked up before, this is also why a quick deletion function is available for removing unwanted words more easily.

>Q1: How should it work in the first place? Do I have to press the Forgot/Got it button for each word? Will flashcards then appear so I can repeat words?

You have to press the button. What you see is already the "flashcards" in a list :)

>Q2: Is there a way to show words only from specific book/books?

No, at least not now. Words are words, they don't seem to have too much to do with which book they came from? I don't really know, if more people want this, I can add it in the future.

>Q3: Is there a way to sort words by hits/misses, alphabet?

No. For now you let the plugin tell you which ones to check each time. They are sorted by the time scheduled for reviewing. It's also why I suggested you to delete them all, because the words should be way overdue now and take up the front positions forever, making the ones you interacted with placed in the last page next time you come.

**Reply by ichnilatis-gr** [reply]
> > Q2: Is there a way to show words only from specific book/books?
> 
> No, at least not now. Words are words, they don't seem to have too much to do with which book they came from? I don't really know, if more people want this, I can add it in the future.

I was also wondering if that would be possible. Maybe it would be useful sometimes.

**Comment by mergen3107** [comment]
@weijiuqiao 
Thank you very much for detailed explanation! 
I will try to delete all words and start anew.

For the book-specific words my first use case that sprung into my mind was learning languages. For example, when I would like to focus on words from my German textbook.

**Reply by weijiuqiao** [reply]
Filtering words by books (from the menu or by swiping up) has been added to the newest release. Check it out if you are interested :)

**Comment by c1-g** [comment]
@weijiuqiao  This changes everything. How haven't I noticed this feature sooner. I'm a spaced repetition frantic. May I ask what algorithm are you using to get the due time?

One more thing, I can see the "Save context" option being used as a way to save an ["extract"](https://supermemo.guru/wiki/Extract) to be used in [Incremental reading](https://supermemo.guru/wiki/Incremental_reading). Where a passage of a text (an extract) is revisited over time via spaced repetition to increase understanding and solidify it into memory.

I understand that this is beyond what "Vocabulary Builder" is built for. Is there a way I can "look up" a whole paragraph of a text to use in vocab builder? How can I go visit the paragraph from tapping the context?

**Reply by weijiuqiao** [reply]
Currently it’s a naive algorithm based on the Ebbinghaus curve. However after #10035 is merged, you will be able to set your own intervals. 
I think one can look up a paragraph and have it as the entry? It may not be that convenient to view though. 
Sadly we can’t jump to the location in books.

**Reply by c1-g** [reply]
I was hoping it would be Anki's SM2 but fixed interval is not too bad! The science on what kind of spacing is better is still up for debate.

And yes I found out that you can have a paragraph as an entry by selecting it and choosing the "Dictionary" option.

The location thing is nice to have but isn't really necessary IMO, since a good extract should be an independent text that can be read and understood on its own. You only need to revisit the file when you don't know the context. ~~I don't know how koreader structure the history of its dictionary but maybe there's an additional info to help find the passage like a chapter name?~~ Nope, lookup_history.lua have no such things.

**Comment by linlance** [comment]
In Kobo Aura One,Where to open flash Card?
Only add word to vocabulary_builder.sqlite3，but where to use it?I can't find a menu Item for it.

**Reply by weijiuqiao** [reply]
It's under the top magnifying glass.

**Comment by mallomar** [comment]
A further follow up to this, I see there’s now an export option to Dropbox, but it’s not the standard Dropbox login. Could you explain how the tokens/credentials work?

**Reply by weijiuqiao** [reply]
You mean the cloud synchronization feature? That's based on the CloudStorage functionality, you can check out #9496 for more info.
